x-pack/plugins/security_solution/server/lib/detection_engine/signals/threat_mapping/enrich_signal_threat_matches.ts

```diff
@@ -56,13 +57,18 @@ export const buildMatchedIndicator = ({
       throw new Error(`Expected indicator field to be an object, but found: ${indicator}`);
     }
     const atomic = get(matchedThreat?._source, query.value) as unknown;
-    const type = get(indicator, 'type') as unknown;
     const event = get(matchedThreat?._source, 'event') as unknown;
 
     return {
       ...indicator,
       event,
-      matched: { atomic, field: query.field, id: query.id, index: query.index, type },
+      matched: {
+        atomic,
+        field: query.field,
+        id: query.id,
+        index: query.index,
+        type: ENRICHMENT_TYPES.IndicatorMatchRule,
```

Comment:
I see the answer to one of my previous questions. this works but it adds a bit of cognitive overhead IMHO, it could be better to use a new separate field to carry this information, such as indicatorType

Reply:
You're right that this is a little confusing, but `matched.type` was previously a useless field, wholly redundant with `indicator.type`. ECS has been updated to reflect this new use, and we can sweep these inconsistencies under the rug with the data migration in 7.15.